openapi.yaml

```diff
@@ -16,175 +16,317 @@ host: api.oregonstate.edu
-      operationId: getPets
+        - notes
+      operationId: getNotes
       parameters:
-        - $ref: '#/parameters/authorization'
-        - $ref: '#/parameters/pageNumber'
-        - $ref: '#/parameters/pageSize'
-        - in: query
-          name: species
-          description: Search by pet species
+        - name: studentID
+          in: query
+          type: string
+          required: true
+          description: 9-digit OSU ID of student whose notes will be retrieved
+        - name: q
+          in: query
+          type: string
           required: false
+          description: Query string to be used when searching through the contents of each note
+        - name: sortKey
+          in: query
           type: string
+          required: false
+          default: 'lastModified'
+          description: The key by which to sort the notes in the response
+        - name: sources
+          in: query
+          type: array
+          items:
+            type: string
+            enum: &SOURCES
+              - advisorPortal
+              - banner
+              - degreeWorks
+            collectionFormat: csv
+          required: false
+          description: A comma-separated list of sources to filter by
+        - name: contextTypes
+          in: query
+          type: array
+          items:
+            type: string
+            collectionFormat: csv
+          required: false
+          description: A comma-separated list of context types to filter by
       responses:
         200:
           description: Successful response
           schema:
-            $ref: '#/definitions/PetsResult'
+            $ref: '#/definitions/NotesResult'
         400:
-          description: Bad Request
+          description: Bad request
           schema:
             $ref: '#/definitions/Errors'
         500:
-          description: Internal Server Error
+          description: Internal server error
           schema:
             $ref: '#/definitions/Errors'
-  /pets/{id}:
-    get:
-      summary: Info for a specific pet
+    post:
+      summary: Create a new note
       tags:
-        - pets
-      description: Get pet by unique ID
-      operationId: getPetById
+        - notes
+      operationId: postNotes
       parameters:
-        - $ref: '#/parameters/authorization'
-        - name: id
-          in: path
-          description: Unique Pet ID
-          required: true
-          type: string
+        - in: body
+          name: requestBody
+          schema:
+            type: object
+            properties:
+              note:
+                $ref: '#/definitions/note'
+              studentID:
+                $ref: '#/definitions/studentID'
+              creatorID:
+                $ref: '#/definitions/creatorID'
+              permissions:
+                $ref: '#/definitions/permissions'
+              context:
+                $ref: '#/definitions/context'
+      responses:
+        200:
```

Comment:
Based on [JSON API](https://jsonapi.org/format/#crud-creating-responses-201), I think we should return a `201 Created`:

> If a POST request did not include a Client-Generated ID and the requested resource has been created successfully, the server MUST return a 201 Created status code.
> The response SHOULD include a Location header identifying the location of the newly created resource.
> The response MUST also include a document that contains the primary resource created.
> If the resource object returned by the response contains a self key in its links member and a Location header is provided, the value of the self member MUST match the value of the Location header.